src/pages/SignupPage.jsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react';
+import styles from './SignupPage.module.css';
+import { Input } from '../components/common/input/Input';
+import { Button } from '../components/common/button/button';
+import { useDispatch, useSelector } from 'react-redux';
+import { signup as signupAction } from '../store/user/actions';
+import Logo from '../components/Navigation/Logo';
+
+function SignupPage() {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [repeatPassword, setRepeatPassword] = useState('');
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [age, setAge] = useState('');
+
+    const { loading, error } = useSelector((state) => state.user.signup);
+
+    const dispatch = useDispatch();
+
+    const onSignup = () => {
+        dispatch(
+            signupAction({
+                username,
+                password,
+                repeatPassword,
+                firstName,
+                lastName,
+                age: Number(age),
+            })
+        );
+    };
+
+    return (
+        <div className={styles.main}>
+            <Logo />
+            <h1 className={styles.signupHeader}>Please signup</h1>
+            <Input
+                placeholder="Username"
+                onChange={setUsername}
+                errorMsg={error?.field === 'username' && error.error}
+            />
+            <Input
+                type="password"
+                placeholder="Password"
+                onChange={setPassword}
+                errorMsg={error?.field === 'password' && error.error}
+            />
+            <Input
+                type="password"
+                placeholder="Repeat password"
+                onChange={setRepeatPassword}
+                errorMsg={error?.field === 'repeatPassword' && error.error}
+            />
+            <Input
+                placeholder="First name"
+                onChange={setFirstName}
+                errorMsg={error?.field === 'firstName' && error.error}
+            />
+            <Input
+                placeholder="Last name"
+                onChange={setLastName}
+                errorMsg={error?.field === 'lastName' && error.error}
+            />
+            <Input
+                type="number"
```

Comment:
you are passing input type here, so don't spread it into otherProps it should be declared

Reply:
I'm not sure it's the best approach. What if we have to pass a lot of props into native input (maxLength, etc) without any changes? If we just keep the naming consistent with native input props, we can easily pass all values through `otherProps`

Reply:
done